### PR TITLE
Websocket input

### PR DESCRIPTION
**Pull request by jduthon**

### CODE REVIEW

File: input/index.js [hunk 1]
```diff
@@ -2,5 +2,6 @@ module.exports = {
   http: require('./http'),
   syslog: require('./syslog'),
   file: require('./file'),
-  elasticsearch: require('./elasticsearch')
+  elasticsearch: require('./elasticsearch'),
+  ws: require('./ws')
```
Comment (znarf) [inline input/index.js]: I would name it "websocket" not "ws".

File: input/websocket.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    name: 'WebSocket',
+    start: (pipeline) => {
+      let socket = new WebSocket(address)
+      pipeline.status(null, 'Waiting for connection to ' + address)
+      socket.on('open', _ => {
+        pipeline.status(null, 'Listening to ' + address)
+      })
+      socket.on('message', stringMessage => {
+        const json = JSON.parse(stringMessage)
+        const messages = Array.isArray(json) ? json : [json]
```
Comment (znarf) [inline input/websocket.js]: Let's remove the batch capabilities, doesn't make sense in that context.

**Comment by znarf** [comment]
@jduthon Could you provide documentation and a default example?

In: https://github.com/access-watch/access-watch/blob/master/docs/input.md

And: https://github.com/access-watch/access-watch/blob/master/config/input.js

Let's use the "JSON" parser, in the example. ```format.json.parser()```

File: input/websocket.js [hunk 3]
```diff
@@ -9,18 +9,14 @@ function create ({address, parse}) {
...
-          pipeline.success(log)
-        })
+      socket.on('message', message => {
+        let log
+        try {
+          log = parse(JSON.parse(message))
+        } catch (err) {
+          return pipeline.error(err)
+        }
+        pipeline.success(log)
```
Comment (znarf) [inline input/websocket.js]: I understand that this code is copied from other inputs, but would it make sense to write it like that and change  the style everywhere else?

```
try {
  const log = parse(JSON.parse(message))
  pipeline.success(log)
} catch (err) {
  pipeline.error(err)
}
```
 ```log``` is a constant used only in the block, and there is no purpose returning the pipeline.error. what do you think @benfle ?

File: docs/input.md [hunk 4]
```diff
@@ -91,6 +91,17 @@ The input accepts the following options.
...
 
+### WebSocket
+
+The WebSocket input subscribe to a WebSocket server sending access logs.
+
+The input accepts the following options.
+
+| Attribute | Type   | Required? | Description                                                              |
+| ---       | ---    | ---       | ---                                                                      |
+| address   | string | yes       | The WebSocket address (e.g. 'wss://localhost:3000')                      |
```
Comment (znarf) [inline docs/input.md]: is 'address' the right word for that, do we have alternatives?

File: input/websocket.js (hunk 3 above)
Comment (jduthon) [reply input/websocket.js]: Yup should be the same

File: docs/input.md (hunk 4 above)
Comment (znarf) [reply docs/input.md]: @jduthon That's a solid +1 for address then :-)

**Pull request merged**